azkaban-common/src/main/java/azkaban/jobExecutor/AbstractProcessJob.java

```diff
@@ -81,6 +84,31 @@ protected void resolveProps() {
     jobProps = PropsUtils.resolveProps(jobProps);
   }
 
+  /**
+   * prints the current Job props to the Job log.
+   */
+  protected void logJobProperties() {
+    if (this.jobProps != null &&
+        this.jobProps.getBoolean(JOB_DUMP_PROPERTIES_IN_LOG, false)){
+      try {
+        Map<String,String> flattenedProps = this.jobProps.getFlattened();
```

Comment:
I was just thinking if there's a way to implement this feature (and sort it) without touching the Props.

I think it can be done with:
TreeSet<String> keySet = jobProps.getKeySet();
for( String key : keySet ){
// do the magic
}

A bit less code and a bit less code change?